src/rt/sections_elf_shared.d

```diff
@@ -725,17 +730,29 @@ version (Shared)
             if (dyn.d_tag == DT_STRTAB)
             {
                 version (CRuntime_Musl)
-                    strtab = cast(const(char)*)(object.baseAddress + dyn.d_un.d_ptr); // relocate
+                    enum relocate = true;
                 else version (linux)
-                    strtab = cast(const(char)*)dyn.d_un.d_ptr;
+                {
```

Comment:
Shouldn't this be `CRuntime_Glibc` then, instead of "linux" ?

Reply:
I'm not keen on looking up `CRuntime_UClibc` etc.

Reply:
At least the MIPS part seems to apply to uclibc too: https://cgit.uclibc-ng.org/cgi/cgit/uclibc-ng.git/tree/ldso/include/ldsodefs.h#n12

@ibuclaw: Do you know if there'd be a more robust way to detect this? Like checking the `PF_W` flag of the `PT_DYNAMIC` program header?

Reply:
This is an internal detail, so no I don't think so.  It did come up when the RISCV developers implemented this in gdc because it seems to have been a mistake to copy MIPS, I guess we'll just have to address that hurdle when we get to it.